### ISSUE TITLE
Vine Futures Definition tied to Submission

### ISSUE THREAD

**Issue by BarrySlyDelgado** [cooperative-computing-lab/cctools#3408]
One Issue with the current model of futures is that task definitions are tied to submission. That is, a tak is submitties at the same time the future is created. The following is an example:

`s = vine.Executor()`
`f = s.submit(func, a, b)` submit creates and submits the task
`print(f.result())`

Here, the future to the task is returned right after the corresponding task is submitted. Thus, there is not time to add any possible dependincies such as input files or environments.

One thought I had on this was creating a handle to that references the function and adding things there before submission like so:

`s = vine.Executor()`
`future_func = s.function(func)`
`future_func.add_input(filename)`
`future_fuc.set_cores(1)`
`f = s.submit(future_func, a, b)`

Whith this method, things could be added to the task but it is messy.

**Comment by btovar** [comment]
Barry, I think it will be easier if s.submit receives TaskVine tasks. In that way a user can submit both regular tasks or python tasks ( current version or @tphung3 's serialized version). In other words, you already have handle for the task... the task itself.

**Comment by dthain** [comment]
I think Ben has the right idea.  Stick with the abstractions that we have.  And then make shortcut for common use cases.

For example, if you can do this:

````
t = FutureTask(func, a,b)
t.set_cores(3)
. . .
m.submit(t)
```

Then you can still provide this shortcut:

```
m.future( func, a, b )
```

That means this:

```
m.submit( FutureTask(func,a,b) )
```

**Comment by dthain** [comment]
@BarrySlyDelgado I think this is fixed in your current version of Futures, correct?

**Comment by BarrySlyDelgado** [comment]
Yes. Fixed in #3403 .
